compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -2010,9 +2144,8 @@ private void analyzeArgIsolatedness(List<BLangExpression> requiredArgs, List<BLa
 
             analyzeAndSetArrowFuncFlagForIsolatedParamArg(arg);
 
-            if (!Symbols.isFlagOn(arg.getBType().flags, Flags.ISOLATED)) {
-                dlog.error(arg.pos, DiagnosticErrorCode.INVALID_NON_ISOLATED_FUNCTION_AS_ARGUMENT);
-            }
+            handleNonExplicitlyIsolatedArgForIsolatedParam(invocationExpr, arg, expectsIsolation,
```

Comment:
Shall we manually check whether these lines are actually not covered?

Reply:
This PR just refactored the check into a method but seems like these weren't covered. Added tests specifically for this line (a334b49) but the rest seems to be blocked by https://github.com/ballerina-platform/ballerina-lang/issues/31847. Created an issue to track adding tests once that is fixed - https://github.com/ballerina-platform/ballerina-lang/issues/31848.